src/Kernel/client/kernel.ts

```diff
@@ -124,9 +124,23 @@ class Kernel {
         IPython.notebook.kernel.events.on("kernel_ready.Kernel", args => {
             this.requestEcho();
             this.requestClientInfo();
+            this.setupMyMessageListener(); //TODO: call this something else
         });
     }
 
+    setupMyMessageListener() {
+        IPython.notebook.kernel.register_iopub_handler(
+            "iqsharp_state_dump",
+            message => {
+                console.log("my message received", message);
+                message.content.amplitudes;
+                var htmlGraph = "hi :) ";
+                document.getElementById(message.content.div_id).innerHTML = htmlGraph;
```

Comment:
You may want to safely handle the case where the `<div>` you're looking for doesn't exist, for example, when `ConfigurationSource.MeasurementDisplayHistogram` is false and you don't create that `<div>` in the first place.

Reply:
Got it, thanks!